secret/store.go

```diff
@@ -0,0 +1,182 @@
+package secret
+
+import (
+	"fmt"
+	"github.com/banzaicloud/bank-vaults/vault"
+	"github.com/banzaicloud/pipeline/config"
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/pkg/errors"
+	"github.com/satori/go.uuid"
+	"github.com/sirupsen/logrus"
+)
+
+var logger *logrus.Logger
+var Store *SecretStore
+
+const (
+	Amazon     = "AMAZON_SECRET"
+	Azure      = "AZURE_SECRET"
+	Google     = "GOOGLE_SECRET"
+	General    = "GENERAL_SECRET"
+	Kubernetes = "KUBERNETES_SECRET"
+)
+
+func init() {
+	logger = config.Logger()
+	Store = newVaultSecretStore()
+}
+
+type SecretStore struct {
+	client  *vault.Client
+	logical *vaultapi.Logical
+}
+
+type CreateSecretResponse struct {
+	Name       string `json:"name" binding:"required"`
+	SecretType string `json:"type" binding:"required"`
+	SecretID   string `json:"secret_id"`
+}
+
+type CreateSecretRequest struct {
+	Name       string            `json:"name" binding:"required"`
+	SecretType string            `json:"type" binding:"required"`
+	Values     map[string]string `json:"values" binding:"required"`
+}
+
+type ListSecretsResponse struct {
+	Secrets []SecretsItemResponse `json:"secrets"`
+}
+
+type SecretsItemResponse struct {
+	ID         string            `json:"id"`
+	Name       string            `json:"name"`
+	SecretType string            `json:"type"`
+	Values     map[string]string `json:"-"`
+}
+
+func newVaultSecretStore() *SecretStore {
+	role := "pipeline"
+	client, err := vault.NewClient(role)
+	if err != nil {
+		panic(err)
+	}
+	logical := client.Vault().Logical()
+	return &SecretStore{client: client, logical: logical}
+}
+
+func GenerateSecretID() string {
+	log := logger.WithFields(logrus.Fields{"tag": "Secret"})
+	log.Debug("Generating secret id")
+	return uuid.NewV4().String()
+}
+
+var DefaultRules = map[string][]string{
+	Amazon: {
+		"AWS_ACCESS_KEY_ID",
+		"AWS_SECRET_ACCESS_KEY",
+	},
+	Azure: {
+		"AZURE_CLIENT_ID",
```

Comment:
These literals should be factored out in a shared places as it used in multiple places